app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/DragAndDropSortInteractionViewModel.kt

```diff
@@ -223,24 +241,41 @@ class DragAndDropSortInteractionViewModel private constructor(
       contentIdHtmlMap: Map<String, String>,
       choiceStrings: List<SubtitledHtml>,
       dragAndDropSortInteractionViewModel: DragAndDropSortInteractionViewModel,
-      resourceHandler: AppLanguageResourceHandler
+      resourceHandler: AppLanguageResourceHandler,
+      listOfSetsOfTranslatableHtmlContentIds: ListOfSetsOfTranslatableHtmlContentIds
     ): MutableList<DragDropInteractionContentViewModel> {
-      return choiceStrings.mapIndexed { index, subtitledHtml ->
-        DragDropInteractionContentViewModel(
-          contentIdHtmlMap = contentIdHtmlMap,
-          htmlContent = SetOfTranslatableHtmlContentIds.newBuilder().apply {
-            addContentIds(
-              TranslatableHtmlContentId.newBuilder().apply {
-                contentId = subtitledHtml.contentId
-              }
+      return if (listOfSetsOfTranslatableHtmlContentIds.contentIdListsCount > 0) {
```

Comment:
These changes could be consolidated, I think, into just the first if-check block. The main difference between the first & second block is that we have the possibility for more than one content ID in the first block, and we could instead convert the choice items into content IDs objects, e.g. something like:

```kotlin
val selectedChoices = if (listOfSetsOfTranslatableHtmlContentIds.contentIdLists.isEmpty()) {
  choiceStrings.map {
    listOf(
      TranslatableHtmlContentId.newBuilder().apply {
        contentId = subtitledHtml.contentId
      }
    )
  }
} else listOfSetsOfTranslatableHtmlContentIds.contentIdListsList.map { it.contentIdsList }

... then use selectedChoices.mapIndexed{}...
```

Reply:
Updated